addons/mail/static/src/messaging/entity/message/message.js

```diff
@@ -0,0 +1,443 @@
+        }
+
+        /**
+         * @returns {string}
+         */
+        get prettyBody() {
+            let prettyBody;
+            for (const emoji of emojis) {
+                const { unicode } = emoji;
+                const regexp = new RegExp(
+                    `(?:^|\\s|<[a-z]*>)(${unicode})(?=\\s|$|</[a-z]*>)`,
+                    "g"
+                );
+                const originalBody = this.body;
+                prettyBody = this.body.replace(
+                    regexp,
+                    ` <span class="o_mail_emoji">${unicode}</span> `
+                );
+                // Idiot-proof limit. If the user had the amazing idea of
+                // copy-pasting thousands of emojis, the image rendering can lead
+                // to memory overflow errors on some browsers (e.g. Chrome). Set an
+                // arbitrary limit to 200 from which we simply don't replace them
+                // (anyway, they are already replaced by the unicode counterpart).
+                if (_.str.count(prettyBody, "o_mail_emoji") > 200) {
+                    prettyBody = originalBody;
+                }
+            }
+            // add anchor tags to urls
+            return parseAndTransform(prettyBody, addLink);
+        }
+
+        replyTo() {
+            const discuss = this.env.messaging.discuss;
+            if (!discuss.isOpen) {
+                return;
+            }
+            if (discuss.replyingToMessage === this) {
+                discuss.clearReplyingToMessage();
+            } else {
+                discuss.link({ replyingToMessage: this });
+            }
+        }
+
+        /**
+         * Toggle check state of this message in the context of the provided
+         * thread and its stringifiedDomain.
+         *
+         * @param {mail.messaging.entity.Thread} thread
+         * @param {string} threadStringifiedDomain
+         */
+        toggleCheck(thread, threadStringifiedDomain) {
+            const threadCache = thread.cache(threadStringifiedDomain);
+            if (threadCache.checkedMessages.includes(this)) {
+                threadCache.unlink({ checkedMessages: this });
+            } else {
+                threadCache.link({ checkedMessages: this });
+            }
+        }
+
+        /**
+         * Toggle the starred status of the provided message.
+         */
+        async toggleStar() {
+            await this.env.rpc({
+                model: 'mail.message',
+                method: 'toggle_message_starred',
+                args: [[this.id]]
+            });
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        _createInstanceLocalId(data) {
+            return `${this.constructor.localId}_${data.id}`;
+        }
```

Comment:
If a local id is a technical detail, shouldn't all entities get a default and irrelevant name coming from the parent (uniqueId) instead of trying to have a relevant name based on data like this?

Reply:
This is really useful for debug at any given time.
Definitely would keep all of this because this is a huge pain to debug the store without that.